nrf_modem/doc/at_interface.rst

```diff
@@ -312,11 +312,16 @@ The following snippet shows how to setup an AT notification handler:
 		return 0;
 	}
 
-.. note::
+The callback is invoked in an interrupt context.
+The user is responsible for rescheduling the processing of AT notifications as appropriate.
+
+In |NCS|, the :ref:`at_monitor_readme` library takes care of dispatching notifications to different parts of the application.
 
-   The callback is invoked in an interrupt context.
-   The user is responsible for rescheduling the processing of AT notifications as appropriate.
-   In |NCS|, the :ref:`at_monitor_readme` library and the :ref:`nrf_modem_lib_readme` takes care of the rescheduling.
+.. important::
+   In NCS applications, many libraries use the :ref:`at_monitor_readme` library to register their own callback with the Modem library using the :c:func:`nrf_modem_at_notif_handler_set` function.
+   If you are building an NCS application, do not use the :c:func:`nrf_modem_at_notif_handler_set` function to register your callback.
```

Comment:
Can we add something like this into the description of `nrf_modem_at_notif_handler_set`?
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/nrf_modem/doc/api.html#c.nrf_modem_at_notif_handler_set

Reply:
I would rather not add it there, because that header comes from libmodem and it shouldn't contain things which relate to NCS since the library could be used without NCS. In that case the documentation would not apply :/

Reply:
I understand and then I don't :) I mean generally people develop modules that could be used without its intended main "client" but in practice it'll never be used without it. And we try to keep things general, which prevents us from adding the necessary documentation that we'd like to add. But that's the way it goes :)